rest/server_context.go

```diff
@@ -86,36 +85,38 @@ func NewServerContext(config *ServerConfig) *ServerContext {
 		sc.startStatsReporter()
 	}
 
-	if config.Replications != nil {
```

Comment:
This diff looks pretty nasty, but I've essentially just pulled this out into a separate method, removed the now-redundant sleep, and added a log for any errors returned.